identity/backend/src/main/java/io/camunda/identity/rolemanagement/service/RoleService.java

```diff
@@ -0,0 +1,110 @@
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.StringUtils;
+
+@Service
+@Transactional
+public class RoleService {
+  private final RoleRepository roleRepository;
+
+  public RoleService(final RoleRepository roleRepository) {
+    this.roleRepository = roleRepository;
+  }
+
+  public Role createRole(final Role role) {
+    if (!StringUtils.hasText(role.getName())) {
+      throw new RuntimeException("role.notValid");
+    }
+
+    if (role.getPermissions() == null) {
+      role.setPermissions(new HashSet<>());
+    }
+
+    return roleRepository.save(role);
+  }
+
+  public void deleteRoleByName(final String roleName) {
+    roleRepository.deleteById(roleName);
+  }
+
+  public Role findRoleByName(final String roleName) {
+    return roleRepository
+        .findById(roleName)
+        .orElseThrow(() -> new RuntimeException("role.notFound"));
+  }
+
+  public List<Role> findAllRoles() {
+    return roleRepository.findAll();
+  }
+
+  public Role updateRole(final String roleName, final Role role) {
+    if (!StringUtils.hasText(role.getName())) {
+      throw new RuntimeException("role.notValid");
+    }
+
+    if (!Objects.equals(roleName, role.getName())) {
+      throw new RuntimeException("role.notFound");
+    }
+
+    if (role.getPermissions() == null) {
+      role.setPermissions(new HashSet<>());
+    }
+
+    final Role existingRole = findRoleByName(roleName);
+
+    existingRole.setDescription(role.getDescription());
+    existingRole.setPermissions(role.getPermissions());
+
+    return roleRepository.save(existingRole);
+  }
+
+  public List<PermissionEnum> findAllPermissionsOfRole(final String roleName) {
+    final Role role = findRoleByName(roleName);
+    return role.getPermissions().stream().toList();
+  }
+
+  public void assignPermissionToRole(final String roleName, final String permissionName) {
+    final PermissionEnum permission =
+        findPermissionByName(permissionName)
+            .orElseThrow(() -> new RuntimeException("permission.notFound"));
+    final Role role = findRoleByName(roleName);
+
+    role.getPermissions().add(permission);
+    roleRepository.save(role);
+  }
+
+  public void unassignPermissionFromRole(final String roleName, final String permissionName) {
+    final PermissionEnum permission =
+        findPermissionByName(permissionName)
+            .orElseThrow(() -> new RuntimeException("permission.notFound"));
+    final Role role = findRoleByName(roleName);
+    role.getPermissions().remove(permission);
+    roleRepository.save(role);
+  }
+
+  private Optional<PermissionEnum> findPermissionByName(final String permissionName) {
+    Optional<PermissionEnum> permission;
+    try {
+      permission = Optional.of(PermissionEnum.valueOf(permissionName));
+    } catch (final IllegalArgumentException e) {
+      permission = Optional.empty();
+    }
+    return permission;
+  }
```

Comment:
I don't think that we need these methods right now since update role also updates the assigned permissions. Let's make it simpler and remove them.